model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -0,0 +1,136 @@
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.OrganizationModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.jpa.entities.OrganizationEntity;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.organization.OrganizationProvider;
+
+public class JpaOrganizationProvider implements OrganizationProvider {
+
+    private final EntityManager em;
+
+    public JpaOrganizationProvider(KeycloakSession session) {
+        JpaConnectionProvider jpaProvider = session.getProvider(JpaConnectionProvider.class);
+        this.em = jpaProvider.getEntityManager();
+    }
+
+    @Override
+    public OrganizationModel createOrganization(RealmModel realm, String name) {
+        throwExceptionIfRealmIsNull(realm);
+        OrganizationEntity entity = new OrganizationEntity();
+
+        entity.setId(KeycloakModelUtils.generateId());
+        entity.setRealm(realm.getId());
+        entity.setName(name);
+
+        em.persist(entity);
+
+        return new OrganizationAdapter(entity);
+    }
+
+    @Override
+    public void updateOrganization(RealmModel realm, OrganizationModel organization) {
+        throwExceptionIfRealmIsNull(realm);
+        throwExceptionIfOrganizationIsNull(organization);
+        OrganizationAdapter adapter = getAdapter(organization.getId());
+        throwExceptionIfOrganizationIsNull(adapter);
+
+        adapter.setName(organization.getName());
+        em.persist(adapter.toEntity());
+    }
+
+    @Override
+    public boolean removeOrganization(RealmModel realm, OrganizationModel organization) {
+        throwExceptionIfRealmIsNull(realm);
+        throwExceptionIfOrganizationIsNull(organization);
+        OrganizationAdapter toRemove = getAdapter(organization.getId());
+        throwExceptionIfOrganizationIsNull(toRemove);
+
+        if (!toRemove.getRealm().equals(realm.getId())) {
+            throw new IllegalArgumentException("Organization [" + organization.getId() + " does not belong to realm [" + realm.getId() + "]");
+        }
+
+        em.remove(toRemove.toEntity());
+
+        return true;
+    }
+
+    @Override
+    public void removeOrganizations(RealmModel realm) {
+        throwExceptionIfRealmIsNull(realm);
+        Query query = em.createNamedQuery("deleteByRealm");
+
+        query.setParameter("realm", realm.getId());
+
+        query.executeUpdate();
+    }
+
+    @Override
+    public OrganizationModel getOrganization(String id) {
+        return getAdapter(id);
+    }
+
+    @Override
+    public Stream<OrganizationModel> getOrganizationsStream(RealmModel realm) {
+        TypedQuery<String> query = em.createNamedQuery("getByRealm", String.class);
+
+        query.setParameter("realm", realm.getId());
+
+        return closing(query.getResultStream().map(this::getAdapter));
+    }
+
+    @Override
+    public void close() {
+
+    }
+
+    private OrganizationAdapter getAdapter(String id) {
```

Comment:
This method seems to be redundant. It could be replaced by `getOrganizationById` (in current version it is called `getOrganization`).

Reply:
Thinking more, I'll keep it for now. We can review that later.